server/src/views/frame.js

```diff
@@ -7,7 +7,70 @@ const { ProfileButton } = require("./profile");
 const { addReactScripts } = require("../reactutils");
 const { gaActivation } = require("../ga-activation");
 
+class ShareButtons extends React.Component {
+  onClickShareButton(whichButton) {
+    //console.log("onClickShareButton", whichButton);
+    // FIXME implement analytics tracking here
+  }
+
+  onClickCopyButton(e) {
+    let target = e.target;
+    target.previousSibling.select();
+    document.execCommand("copy");
+    target.textContent = "Copied";
+    setTimeout(function() {
+      target.textContent = "Copy";
+    }, 1000);
```

Comment:
This could use `this.setState({copyText: "Copied"})` and a timeout to undo that state, and then just render the button using `<button>{this.state.copyText}</button>`